src/builder.ts

```diff
@@ -49,6 +49,17 @@ export default class Builder {
 	}
 
 	/**
+	 * Initialise the builder class with an already created Dockerode object,
+	 * useful for when a connection has already been setup
+	 */
+	public static fromDockerode(docker: Dockerode): Builder {
```

Comment:
Looks good, just wondering if you can make a `private setDocker(docker: Dockerode)` method that sets the `docker` and creates the async version to avoid duplicating that. If you think this is better just go ahead with this.